include/deal.II/lac/petsc_ts.h

```diff
@@ -583,12 +576,54 @@ namespace PETScWrappers
      */
     std::function<IndexSet()> algebraic_components;
 
+    /**
+     * Callback for the recoverable action from a TimeStepper::function.
+     *
+     * Deal.II internally communicates the necessary actions to the PETSc
+     * object. This additional callback returns control to the user
+     * for additional cleanup.
+     */
+    std::function<void()> recoverable_action_function;
+
+    /**
+     * Callback for the recoverable action from a TimeStepper::jacobian
+     * or TimeStepper::setup_jacobian.
+     *
+     * Deal.II internally communicates the necessary actions to the PETSc
+     * object. This additional callback returns control to the user
+     * for additional cleanup.
+     */
+    std::function<void()> recoverable_action_jacobian;
+
+    /**
+     * Callback for the recoverable action from a
+     * TimeStepper::solve_with_jacobian.
+     *
+     * Deal.II internally communicates the necessary actions to the PETSc
+     * object. This additional callback returns control to the user
+     * for additional cleanup.
+     */
+    std::function<void()> recoverable_action_solve_with_jacobian;
+
+    /**
+     * Callback for the recoverable action from a
+     * TimeStepper::monitor.
+     *
+     * Deal.II internally communicates the necessary actions to the PETSc
+     * object. This additional callback returns control to the user
+     * for additional cleanup.
+     */
+    std::function<void()> recoverable_action_monitor;
+
```

Comment:
I don't think you need those any more, right? If so, please just delete.

Reply:
yes, they are leftovers. I will remove them